src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -150,15 +150,20 @@ public override PSThreadOptions ThreadOptions
                     {
                         if (this.RunspaceStateInfo.State != RunspaceState.BeforeOpen)
                         {
-#if CORECLR                 // No ApartmentState.STA Support In CoreCLR
-                            bool allowed = value == PSThreadOptions.ReuseThread;
-#else
-                            // if the runspace is already opened we only allow changing the options if
-                            // the apartment state is MTA and the new value is ReuseThread
-                            bool allowed = (this.ApartmentState == ApartmentState.MTA || this.ApartmentState == ApartmentState.Unknown) // Unknown is the same as MTA
-                                           &&
-                                           value == PSThreadOptions.ReuseThread;
-#endif
+                            bool allowed = false;
```

Comment:
I know this is the existing logic, but it's a pretty complicated way to calculate an if condition.

```csharp
bool areThreadOptionsValid = value == PSThreadOptions.ReuseThread;
if (!(Platform.IsNanoServer || Platform.IsIoT))
{
    areThreadOptionsValid &= this.ApartmentState == ApartmentState.MTA || this.ApartmentState == ApartmentStatel.Unknown;
}

if (!areThreadOptionsValid)
{
    ...
}
```

Or ideally:

```csharp
lock (this.SyncRoot)
{
    // Might be able to put this outside the lock even???
    if (value == _createThreadOptions)
    {
        return;
    }

    if (IsInvalidThreadOptionsConfiguration(value))
    {
        throw new InvalidOperationException(StringUtil.Format(RunspaceStrings.InvalidThreadOptionsChange));
    }

    _createThreadOptions = value;
}

...

private boolean IsInvalidThreadOptionsConfiguration(PSThreadOptions options)
{
    if (this.RunspaceStateInfo.State == RunspaceState.BeforeOpen)
    {
        return true;
    }

    // If the runspace is already opened, we only allow changing options when:
    //  - The new value is ReuseThread, and
    //  - The apartment state is MTA

    if (options != PSThreadOptions.ReuseThread)
    {
        return false;
    }

    if (Platform.IsNanoServer || Platform.IsIoT)
    {
        // NanoServer and IoT are always MTA
        return true;
    }

    return this.ApartmentState == ApartmentState.MTA
        || this.ApartmentState == ApartmentState.Unknown; // Unknown is the same as MTA
}
```

Reply:
Will change